client/ayon_maya/plugins/publish/collect_look.py

```diff
@@ -655,6 +662,55 @@ def collect_resources(self, node):
                 "color_space": color_space
             }
 
+    def collect_texture_reference_object_inputs(self, instance):
+        """Collect the inputs for all nodes in the input_SET"""
+
+        # Get the input meshes information
+        input_content = instance.data["setMembers"]
+
+        # Include children
+        input_content += cmds.listRelatives(input_content,
+                                            allDescendents=True,
+                                            fullPath=True) or []
+
+        # Ignore intermediate objects
+        input_content = cmds.ls(input_content, long=True, noIntermediate=True)
+        if not input_content:
+            return []
+
+        # Store all connections
+        connections = cmds.listConnections(input_content,
+                                           source=True,
+                                           destination=False,
+                                           connections=True,
+                                           # Only allow inputs from dagNodes
+                                           # (avoid display layers, etc.)
+                                           type="dagNode",
+                                           plugs=True) or []
+        connections = cmds.ls(connections, long=True)      # Ensure long names
```

Comment:
The instance should already have all the members (including the hierarchy collected)
```suggestion
        # Get meshes
        meshes = cmds.ls(
            list(instance),
            noIntermediate=True,
            long=True
        )
        
        # Get input connections to `referenceObject` attribute
        attrs = [f"{mesh}.referenceObject" for mesh in meshes]
        connections = cmds.listConnections(
            attrs,
            destination=False,
            connections=True,
            # Only allow input meshes
            type="mesh",
            plugs=True) or []
        )
        connections = cmds.ls(connections, long=True)      # Ensure long names
```

Note that [Maya 2023.1 added `fullNodeName` argument to `listConnections`](https://help.autodesk.com/cloudhelp/2023/ENU/Maya-Tech-Docs/CommandsPython/listConnections.html#flagfullNodeName) but we can't rely on that yet 😢